File: sensate-service/Models/Measurement.cs
/*
 * Measurement model
 *
 * @author: Michel Megens
 * @email:  dev@bietje.net
 */

using System;

using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

using Newtonsoft.Json;
using SensateService.Converters;

namespace SensateService.Models
{
	public class Measurement
	{
		[BsonId, BsonRequired, JsonConverter(typeof(ObjectIdJsonConverter))]
		public ObjectId InternalId {get;set;}
		[BsonRequired]
		public decimal Data {get;set;}
		public double Longitude {get;set;}
		public double Latitude {get;set;}
		[BsonRequired]
		public DateTime CreatedAt {get;set;}
		[BsonRequired]
		public ObjectId CreatedBy {get;set;}

		public Measurement()
		{
			this.InternalId = ObjectId.Empty;
			this.CreatedBy = ObjectId.Empty;
		}

		public string ToJson()
		{
			return JsonConvert.SerializeObject(this);
		}
	}
}

File: LiveDataHandler/src/models/measurement.ts
/*
 * Measurement model definition.
 *
 * @author Michel Megens
 * @email  michel.megens@sonatolabs.com
 */

import { Types } from "mongoose";

export interface IDataPoint {
    Name: string;
    Value: number;
    Unit? : string;
};

export class Measurement {
    Data: [IDataPoint];
    Longitude: number;
    Latitude: number;
    CreatedAt: number;
    CreatedBy: Types.ObjectId;
}
